scripts/eosio_build_ubuntu.sh

```diff
@@ -1,430 +1,262 @@
+	&& rm -f mongodb-linux-x86_64-ubuntu$OS_MAJ$OS_MIN-$MONGODB_VERSION.tgz \
+	&& cp -f $CURRENT_DIR/scripts/mongod.conf $MONGODB_CONF \
+	&& mkdir -p $MONGODB_DATA_LOCATION \
+	&& rm -rf $MONGODB_LINK_LOCATION \
+	&& rm -rf $BIN_LOCATION/mongod \
+	&& ln -s $MONGODB_ROOT $MONGODB_LINK_LOCATION \
+	&& ln -s $MONGODB_LINK_LOCATION/bin/mongod $BIN_LOCATION/mongod \
+	|| exit 1
+	printf " - MongoDB successfully installed @ ${MONGODB_ROOT} (Symlinked to ${MONGODB_LINK_LOCATION}).\\n"
+else
+	printf " - MongoDB found with correct version @ ${MONGODB_ROOT} (Symlinked to ${MONGODB_LINK_LOCATION}).\\n"
+fi
+printf "Checking MongoDB C driver installation...\\n"
+if [ ! -d $MONGO_C_DRIVER_ROOT ]; then
+	printf "Installing MongoDB C driver...\\n"
+	curl -LO https://github.com/mongodb/mongo-c-driver/releases/download/$MONGO_C_DRIVER_VERSION/mongo-c-driver-$MONGO_C_DRIVER_VERSION.tar.gz \
+	&& tar -xf mongo-c-driver-$MONGO_C_DRIVER_VERSION.tar.gz \
+	&& cd mongo-c-driver-$MONGO_C_DRIVER_VERSION \
+	&& mkdir -p cmake-build \
+	&& cd cmake-build \
+	&& cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$HOME -DENABLE_BSON=ON -DENABLE_SSL=OPENSSL -DENABLE_AUTOMATIC_INIT_AND_CLEANUP=OFF -DENABLE_STATIC=ON .. \
+	&& make -j"${JOBS}" \
+	&& make install \
+	&& cd ../.. \
+	&& rm mongo-c-driver-$MONGO_C_DRIVER_VERSION.tar.gz \
+	|| exit 1
+	printf " - MongoDB C driver successfully installed @ ${MONGO_C_DRIVER_ROOT}.\\n"
+else
+	printf " - MongoDB C driver found with correct version @ ${MONGO_C_DRIVER_ROOT}.\\n"
+fi
+printf "Checking MongoDB C++ driver installation...\\n"
+if [ ! -d $MONGO_CXX_DRIVER_ROOT ]; then
+	printf "Installing MongoDB C++ driver...\\n"
+	curl -L https://github.com/mongodb/mongo-cxx-driver/archive/r$MONGO_CXX_DRIVER_VERSION.tar.gz -o mongo-cxx-driver-r$MONGO_CXX_DRIVER_VERSION.tar.gz \
+	&& tar -xzvf mongo-cxx-driver-r${MONGO_CXX_DRIVER_VERSION}.tar.gz \
+	&& cd mongo-cxx-driver-r$MONGO_CXX_DRIVER_VERSION/build \
+	&& cmake -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$HOME .. \
+	&& make -j"${JOBS}" VERBOSE=1 \
+	&& make install \
+	&& cd ../.. \
+	&& rm -f mongo-cxx-driver-r$MONGO_CXX_DRIVER_VERSION.tar.gz \
+	|| exit 1
+	printf " - MongoDB C++ driver successfully installed @ ${MONGO_CXX_DRIVER_ROOT}.\\n"
+else
+	printf " - MongoDB C++ driver found with correct version @ ${MONGO_CXX_DRIVER_ROOT}.\\n"
+fi
+
+
+printf "\\n"
+
+
+printf "Checking LLVM 4 support...\\n"
+if [ ! -d $LLVM_ROOT ]; then
+	ln -s /usr/lib/llvm-4.0 $LLVM_ROOT \
+	|| exit 1
+	printf " - LLVM (WASM compiler) successfully linked from /usr/lib/llvm-4.0 to ${LLVM_ROOT}\\n"
+else
+	printf " - LLVM (WASM compiler) found @ ${LLVM_ROOT}.\\n"
+fi
+
+cd ..
+printf "\\n"
 
-	function print_instructions()
-	{
-		printf '\n\texport PATH=${HOME}/opt/mongodb/bin:$PATH\n'
-		printf "\\t%s -f %s &\\n" "$( command -v mongod )" "${MONGOD_CONF}"
-		printf "\\tcd %s; make test\\n\\n" "${BUILD_DIR}"
+function print_instructions()
```

Comment:
IMO asking users to do these instructions is a support nightmare. I don't know about the mongo stuff, but for cmake, cmake options, and ld_library_path there are probably better ways of going about this

Reply:
To make mac and linux more uniform (for supportability) I had to use $HOME for the build/link locations. Unfortunately, this required manipulating ENV (LD/CPATH, etc). This sets us up to support situations like the one with fedora27 where the include directory is non-standard. I'm open to suggestions of course.

Reply:
If a user is using the build script, the build script is the sole method by which cmake is called and cmake is configured. There shouldn't be any need to have things like CMAKE_MODULE_PATH or CPATH in the environment or have cmake in path because any sort of configuration of that nature can be done entirely in the script.

What is LD_LIBRARY_PATH used/needed for? It sounds like we're building an executable that pulls in shared libraries that we also build but I'm not sure what that would be, mongo maybe?

Reply:
It seemed as if `make test` after build was failing without having the llvm/lib stuff loaded in the user's environment. This was due to the $HOME/opt/llvm path.

Reply:
What was the error? I don't believe anything in ~/opt/llvm would be used post build as long as everything was still being linked statically

Reply:
I'm pulling the llvm/lib right now and running some tests. I'll let you know how it goes